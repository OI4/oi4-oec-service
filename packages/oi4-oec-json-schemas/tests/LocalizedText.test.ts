import { matchersWithOptions } from 'jest-json-schema'
import schema from '../schemas/LocalizedText.schema.json'
import localeSchema from '../schemas/constants/locale.pattern.schema.json'

import validObjs from './__fixtures__/LocalizedText_valid.json'
import invalidObjs from './__fixtures__/LocalizedText_invalid.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  }, (ajv) => {
    ajv.addSchema(localeSchema)
  })
)

describe('LocalizedText schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it.each(validObjs as [])(
    '(%#) match valid config object to schema -> %s',
    (_name: string, obj) => {
      expect(obj).toMatchSchema(schema)
    }
  )

  it.each(invalidObjs as [])(
    '(%#) match fails for invalid config -> %s',
    (name: string, obj) => {
      expect(obj).not.toMatchSchema(schema)
      expect(() => {
        expect(obj).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(name)
    }
  )
})
