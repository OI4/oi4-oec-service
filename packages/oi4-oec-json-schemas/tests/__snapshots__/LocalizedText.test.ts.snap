// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedText schema (0) match fails for invalid config -> empty: empty 1`] = `
"expect(received).toMatchSchema(schema)

received
  should have required property 'locale'
    Path:     LocalizedText.schema.json#/required
  should have required property 'text'
    Path:     LocalizedText.schema.json#/required
"
`;

exports[`LocalizedText schema (1) match fails for invalid config -> Invalid Locale: Invalid Locale 1`] = `
"expect(received).toMatchSchema(schema)

received
  .locale should match pattern \\"^[A-z]{2}(-[A-z]{2,3})?$\\"
    Received: <string> ene
    Path:     LocalizedText.schema.jsonlocale.pattern.schema.json/pattern
"
`;

exports[`LocalizedText schema (2) match fails for invalid config -> Invalid Locale: Invalid Locale 1`] = `
"expect(received).toMatchSchema(schema)

received
  .locale should match pattern \\"^[A-z]{2}(-[A-z]{2,3})?$\\"
    Received: <string> en-USofA
    Path:     LocalizedText.schema.jsonlocale.pattern.schema.json/pattern
"
`;

exports[`LocalizedText schema (3) match fails for invalid config -> Invalid Text: Invalid Text 1`] = `
"expect(received).toMatchSchema(schema)

received
  .text should NOT be shorter than 1 characters
    Received: <empty string>
    Path:     LocalizedText.schema.json#/properties/text/minLength
"
`;

exports[`LocalizedText schema (4) match fails for invalid config -> missing locale: missing locale 1`] = `
"expect(received).toMatchSchema(schema)

received
  should have required property 'locale'
    Path:     LocalizedText.schema.json#/required
"
`;

exports[`LocalizedText schema (5) match fails for invalid config -> empty locale & text: empty locale & text 1`] = `
"expect(received).toMatchSchema(schema)

received
  .locale should match pattern \\"^[A-z]{2}(-[A-z]{2,3})?$\\"
    Received: <empty string>
    Path:     LocalizedText.schema.jsonlocale.pattern.schema.json/pattern
  .text should NOT be shorter than 1 characters
    Received: <empty string>
    Path:     LocalizedText.schema.json#/properties/text/minLength
"
`;

exports[`LocalizedText schema (6) match fails for invalid config -> missing text: missing text 1`] = `
"expect(received).toMatchSchema(schema)

received
  should have required property 'text'
    Path:     LocalizedText.schema.json#/required
"
`;
